### PR TITLE
rbd-mirror: pool watcher should track mirror uuid

### PR DESCRIPTION
**Pull request by dillaman**

### CODE REVIEW

File: src/tools/rbd_mirror/PoolWatcher.cc [hunk 1]
```diff
@@ -95,6 +95,9 @@ void PoolWatcher<I>::init(Context *on_finish) {
     m_on_init_finish = on_finish;
   }
 
+  assert(!m_refresh_in_progress);
+  m_refresh_in_progress = true;
```
Comment (trociny) [inline src/tools/rbd_mirror/PoolWatcher.cc]: @dillaman I am just wondering why do you access m_refresh_in_progress not holding the lock (here and inprocess_refresh_images below)?

**Comment by trociny** [comment]
@dillaman s/tracker/track/ in the PR title?

**Comment by dillaman** [comment]
@trociny update pushed

**Comment by dillaman** [comment]
retest this please

File: src/tools/rbd_mirror/PoolWatcher.cc [hunk 2]
```diff
@@ -495,12 +563,37 @@ template <typename I>
...
+    if (m_mirror_uuid != m_pending_mirror_uuid) {
+      if (!m_mirror_uuid.empty()) {
+        dout(0) << "mirror uuid updated:"
+                << "old=" << m_mirror_uuid << ", "
+                << "new=" << m_pending_mirror_uuid << dendl;
+      }
+
+      mirror_uuid = m_mirror_uuid;
+      removed_image_ids = std::move(m_image_ids);
+      m_image_ids.clear();
```
Comment (dillaman) [inline src/tools/rbd_mirror/PoolWatcher.cc]: @trociny Per the standard, the object is left in an "undefined" state after a move and you are required to put it back into a well-defined state. For STL collections, the "clear" method is used to accomplish that.

**Pull request merged**